x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_setup_state.tsx

```diff
@@ -4,52 +4,91 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { useState, useCallback, useMemo } from 'react';
+import { useState, useCallback, useMemo, useEffect } from 'react';
 
 import { isExampleDataIndex } from '../../../../common/log_analysis';
+import { ValidationIndicesUIError } from '../../../pages/logs/analysis/setup/initial_configuration_step';
+import { ValidationIndicesError } from '../../../../common/http_api';
+import { useTrackedPromise } from '../../../utils/use_tracked_promise';
+import { callIndexPatternsValidate } from './api/index_patterns_validate';
 
 type SetupHandler = (
   indices: string[],
   startTime: number | undefined,
   endTime: number | undefined
 ) => void;
 
+interface ValidatedIndex {
+  index: string;
+  validation?: ValidationIndicesError;
+}
+
 interface AnalysisSetupStateArguments {
   availableIndices: string[];
   cleanupAndSetupModule: SetupHandler;
   setupModule: SetupHandler;
+  timestampField: string;
 }
 
 type IndicesSelection = Record<string, boolean>;
 
-type ValidationErrors = 'TOO_FEW_SELECTED_INDICES';
-
 const fourWeeksInMs = 86400000 * 7 * 4;
 
 export const useAnalysisSetupState = ({
   availableIndices,
   cleanupAndSetupModule,
   setupModule,
+  timestampField,
 }: AnalysisSetupStateArguments) => {
   const [startTime, setStartTime] = useState<number | undefined>(Date.now() - fourWeeksInMs);
   const [endTime, setEndTime] = useState<number | undefined>(undefined);
 
-  const [selectedIndices, setSelectedIndices] = useState<IndicesSelection>(
-    availableIndices.reduce(
-      (indexMap, indexName) => ({
-        ...indexMap,
-        [indexName]: !(availableIndices.length > 1 && isExampleDataIndex(indexName)),
-      }),
-      {}
-    )
+  // Show an empty list while we run the validation
```

Comment:
https://thecodinglove.com/code-comments

As a matter of fact I did put a spinner afterwards. I'll remove the comment.

Reply:
Right - still, how about using the overlay instead of hiding the checkboxes completely?

Reply:
ahhh! I understand now.

I'm not sure. I think it's better UX to now show the list until we know which indices are valid and which are not. I think the checkboxes should not be marked until we know if the indices are valid. If we show a list with an overlay we need to either
- Select all checkboxes by default, then deselect/disable the invalid ones.
- Deselect all checkboxes by default, then select the valid ones.

Both scenarios feel (for me) like "something is going on, and the state has changed magically because of that". Like, the initial state we show you is actually not valid and we need to do extra work.

With the current scenario it feels more like "something is being loaded, and it shows when it's done".

Reply:
Yes, the checkboxes should probably be deselected and disabled while loading. The overlay would block and mouse interaction.

I can follow your argument, but I experience the visual impression of the sole spinner as jarring. Having the list of indices visible would mean the layout doesn't jump and the get a hint at what is being loaded.

If we only display the loading spinner we at least might consider adding a descriptive text that hints at what will show up.

Reply:
I did a quick test. This is how it looks like:

<img width="841" alt="Screenshot 2019-11-18 at 15 21 52" src="https://user-images.githubusercontent.com/57448/69060674-081f8e00-0a18-11ea-87fe-8fc270d28122.png">

I prefer how it looks with only the spinner, but I don't have a strong opinion about it. If you think this is better I'll commit the change 🚀